google/cloud/firestore_v1/async_vector_query.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from google.api_core import gapic_v1
+from google.api_core import retry_async as retries
+from google.cloud.firestore_v1 import async_document
+from google.cloud.firestore_v1.base_document import DocumentSnapshot
+from google.cloud.firestore_v1.base_query import (
+    BaseQuery,
+    _query_response_to_snapshot,
+    _collection_group_query_response_to_snapshot,
+)
+from google.cloud.firestore_v1.base_vector_query import BaseVectorQuery
+from typing import AsyncGenerator, List, Union, Optional, TypeVar
+
+TAsyncVectorQuery = TypeVar("TAsyncVectorQuery", bound="AsyncVectorQuery")
+
+
+class AsyncVectorQuery(BaseVectorQuery):
+    """Represents an async vector query to the Firestore API."""
+
+    def __init__(
+        self,
+        nested_query: Union[BaseQuery, TAsyncVectorQuery],
+    ) -> None:
+        """Presents the vector query.
+        Args:
+            nested_query (BaseQuery | VectorQuery): the base query to apply as the prefilter.
+        """
+        super(AsyncVectorQuery, self).__init__(nested_query)
+
+    async def get(
+        self,
+        transaction=None,
+        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
+        timeout: Optional[float] = None,
+    ) -> List[DocumentSnapshot]:
+        """Runs the vector query.
+
+        This sends a ``RunQuery`` RPC and returns a list of document messages.
+
+        Args:
+            transaction
+                (Optional[:class:`~google.cloud.firestore_v1.transaction.Transaction`]):
+                An existing transaction that this query will run in.
+                If a ``transaction`` is used and it already has write operations
+                added, this method cannot be used (i.e. read-after-write is not
+                allowed).
+            retry (google.api_core.retry.Retry): Designation of what errors, if any,
+                should be retried.  Defaults to a system-specified policy.
+            timeout (float): The timeout for this request.  Defaults to a
+                system-specified value.
+
+        Returns:
+            list: The vector query results.
+        """
+        stream_result = self.stream(
+            transaction=transaction, retry=retry, timeout=timeout
+        )
+        result = [snapshot async for snapshot in stream_result]
+        return result  # type: ignore
+
+    async def stream(
```

Comment:
I just realized `stream` isn't present in `BaseVectorQuery`, but it is implemented in both `VectorQuery` and `AsyncVectorQuery`. Is that intentional?

Reply:
Good point. Added one.